diff --git a/backends/fs/emscripten/http-fs.cpp b/backends/fs/emscripten/http-fs.cpp
index 7eecaf22..5e4fbe97 100644
--- a/backends/fs/emscripten/http-fs.cpp
+++ b/backends/fs/emscripten/http-fs.cpp
@@ -92,6 +92,7 @@ EM_ASYNC_JS(char *, _httpFsFetchIndex, (const char *url), {
 	globalThis['httpFsIndexCache'] = globalThis['httpFsIndexCache'] || {};
 	returnString = "";
 	url = UTF8ToString(url);
+	if (Module['httpFsBaseUrl'] && url.startsWith('/')) url = Module['httpFsBaseUrl'] + url;
 	console.debug("Downloading %s", url);
 	if (globalThis['httpFsIndexCache'][url]) {
 		console.debug("Cache hit for %s", url);
@@ -116,6 +117,7 @@ EM_ASYNC_JS(char *, _httpFsFetchIndex, (const char *url), {
 EM_ASYNC_JS(bool, _httpFsFetchFile, (const char *url, byte *dataPtr, int dataSize), {
 	returnBytes = new Uint8Array();
 	url = UTF8ToString(url);
+	if (Module['httpFsBaseUrl'] && url.startsWith('/')) url = Module['httpFsBaseUrl'] + url;
 	console.debug("Downloading %s", url);
 	try {
 		const response = await fetch(url);
